src/install-peerdeps.js

```diff
@@ -106,17 +106,20 @@ function spawnInstall(command, args) {
  * Gets the contents of the package.json for a package at a specific version
  * @param {Object} requestInfo - information needed to make the request for the data
  * @param {string} requestInfo.packageName - the name of the package
+ * @param {Boolean} requestInfo.noRegistry - Gets the package dependencies list from the local node_modules instead of remote registry
  * @param {string} requestInfo.registry - the URI of the registry on which the package is hosted
- * @param {Boolean} onlyPeers - true if only the peers have been requested to be installed. In this case, check for the package to already have been installed.
- * @param {string} version - the version (or version tag) to attempt to install. Ignored if an installed version of the package is found in node_modules.
+ * @param {string} requestInfo.version - the version (or version tag) to attempt to install. Ignored if an installed version of the package is found in node_modules.
  * @returns {Promise<Object>} - a Promise which resolves to the JSON response from the registry
  */
-function getPackageJson(
-  { packageName, registry, auth, proxy },
-  onlyPeers,
+function getPackageJson({
+  packageName,
+  noRegistry,
+  registry,
+  auth,
+  proxy,
   version
-) {
-  if (onlyPeers) {
+}) {
+  if (noRegistry) {
```

Comment:
this just means it checks node_modules; wouldn't instead we want the change to be from "check the default registry only" to "check the appropriate registry for the package"?

Reply:
In the subsequent PR I'm stripping out the registry stuff entirely because it can be offloaded to the package manager. But the option `no-registry` is preserved. In that PR whether or not it's the "appropriate" registry or not becomes irrelevant since this package stops being responsible for dealing with registry management.

For this PR I wanted to clarify the behavior since it's a breaking change on how onlyPeers behaved that could have impact to some user's workflow.